NitroxClient/Communication/Packets/Processors/FootstepPacketProcessor.cs

```diff
@@ -0,0 +1,66 @@
+using FMOD;
+using FMOD.Studio;
+using FMODUnity;
+using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxClient.GameLogic;
+using NitroxClient.GameLogic.FMOD;
+using NitroxModel.Packets;
+
+namespace NitroxClient.Communication.Packets.Processors;
+
+public class FootstepPacketProcessor : ClientPacketProcessor<FootstepPacket>
+{
+    private readonly PlayerManager remotePlayerManager;
+    private readonly FootstepSounds localFootstepSounds;
+    private PARAMETER_ID fmodIndexSpeed = FMODUWE.invalidParameterId;
+    private readonly float footstepAudioRadius = 20f;
+    private readonly float footstepAudioMaxVolume = 0.5f;
+
+    public FootstepPacketProcessor(PlayerManager remotePlayerManager)
+    {
+        this.remotePlayerManager = remotePlayerManager;
+        this.localFootstepSounds = Player.mainObject.GetComponent<FootstepSounds>();
+    }
+
+    public override void Process(FootstepPacket packet)
+    {
+        var player = remotePlayerManager.Find(packet.playerID);
+        if (!player.HasValue)
+        {
+            return;
+        }
+        else
+        {
+            FMODAsset asset;
+            switch (packet.assetIndex)
+            {
+                case FootstepPacket.StepSounds.PRECURSOR_STEP_SOUND:
+                    asset = localFootstepSounds.precursorInteriorSound;
+                    break;
+                case FootstepPacket.StepSounds.METAL_STEP_SOUND:
+                    asset = localFootstepSounds.metalSound;
+                    break;
+                case FootstepPacket.StepSounds.LAND_STEP_SOUND:
+                    asset = localFootstepSounds.landSound;
+                    break;
+                default:
+                    asset = null;
+                    break;
+            }
```

Comment:
Could be changed to a switch expression if you want/find it better:

```
            FMODAsset asset = packet.assetIndex switch
            {
                FootstepPacket.StepSounds.PRECURSOR_STEP_SOUND => localFootstepSounds.precursorInteriorSound,
                FootstepPacket.StepSounds.METAL_STEP_SOUND => localFootstepSounds.metalSound,
                FootstepPacket.StepSounds.LAND_STEP_SOUND => localFootstepSounds.landSound,
                _ => null
            };
```

Reply:
Done